test/jdk/java/foreign/valist/VaListTest.java

```diff
@@ -183,7 +186,8 @@ public static Object[][] sumDoubles() {
         Function<ValueLayout.OfDouble, BiFunction<Integer, VaList, Double>> sumDoublesJavaFact  = layout ->
                 (num, list) -> DoubleStream.generate(() -> list.nextVarg(layout)).limit(num).sum();
         BiFunction<Integer, VaList, Double> sumDoublesNative
-                = MethodHandleProxies.asInterfaceInstance(BiFunction.class, MH_sumDoubles);
+                = MethodHandleProxies.asInterfaceInstance(BiFunction.class,
+                MethodHandles.filterArguments(MH_sumDoubles, 1, VALIST_TO_ADDRESS));
```

Comment:
I suggest maybe moving this adaptation to `linkVaList` (the VaList should always be the last parameter)